packages/GisidaLite/src/legend/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import './legend.css';
+import { legendBlockBuilder } from './utils/helpers';
+
+/** legend data type */
+export interface LegendBlock {
+  color: string;
+  fillWidth: string;
+  label: string;
+}
+/** Legend props */
+export interface LegendProps {
+  legendData: LegendBlock[];
+  legendCreditText?: string;
+}
+
+/** Default props for GisidaLite */
+export const defaultLegendProps: LegendProps = {
```

Comment:
How were these chosen?

Reply:
For grp we are using [ckmeans](https://www.npmjs.com/package/ckmeans) for clustering the data then building the labels based on next and current index. The colors are coming from config and the width is calculated by dividing 100 by colors.length. I think this use case is a bit custom and maybe handling this separately from this package would be ideal

Reply:
How would that look like?

Reply:
Maybe adding an issue that adds utils to build this based on the various codebases would be necessary. Implementing this in a custom way may require changes in the future which may require changing  the code or  reverting back to overriding the legend data for most use cases. The plan I'd go with is create a discussion ticket around this so we can establish a standard approach with which this package can be used across the various projects.

Reply:
I've added this issue here https://github.com/onaio/js-tools/issues/186